File: python/meteroid/models/transaction.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .customer_payment_method_id import CustomerPaymentMethodId
from .payment_method_info import PaymentMethodInfo
from .payment_status_enum import PaymentStatusEnum
from .payment_transaction_id import PaymentTransactionId
from .payment_type_enum import PaymentTypeEnum


@dataclasses.dataclass
class Transaction(BaseModel):
    amount: int

    currency: str

    id: PaymentTransactionId

    payment_type: PaymentTypeEnum

    status: PaymentStatusEnum

    error: t.Optional[str] = None

    payment_method_id: t.Optional[CustomerPaymentMethodId] = None

    payment_method_info: t.Optional[PaymentMethodInfo] = None

    processed_at: t.Optional[datetime] = None

    provider_transaction_id: t.Optional[str] = None

File: go/transaction.go
// this file is @generated
package meteroid

import "time"

type Transaction struct {
	Amount int64 `json:"amount"`

	Currency string `json:"currency"`

	Error *string `json:"error,omitempty"`

	Id PaymentTransactionId `json:"id"`

	PaymentMethodId *CustomerPaymentMethodId `json:"payment_method_id,omitempty"`

	PaymentMethodInfo *PaymentMethodInfo `json:"payment_method_info,omitempty"`

	PaymentType PaymentTypeEnum `json:"payment_type"`

	// RFC 3339 timestamp.
	ProcessedAt *time.Time `json:"processed_at,omitempty"`

	ProviderTransactionId *string `json:"provider_transaction_id,omitempty"`

	Status PaymentStatusEnum `json:"status"`
}
